modules/highgui/src/window_QT.h

```diff
@@ -431,6 +438,55 @@ class OCVViewPort : public ViewPort
 
 #ifdef HAVE_QT_OPENGL
 
+// Use QOpenGLWidget for Qt6 (QGLWidget is deprecated)
+#ifdef HAVE_QT6
+class OpenGlViewPort : public QOpenGLWidget, public OCVViewPort
+{
+public:
+    explicit OpenGlViewPort(QWidget* parent);
+
+    // Note: #define CV_OVERIDE = override (#include < opencv2/core/cvdef.h >)
+    ~OpenGlViewPort() CV_OVERRIDE;
+
+    QWidget* getWidget() CV_OVERRIDE;
+
+    void writeSettings(QSettings& settings) CV_OVERRIDE;
+    void readSettings(QSettings& settings) CV_OVERRIDE;
+
+    double getRatio() CV_OVERRIDE;
+    void setRatio(int flags) CV_OVERRIDE;
+
+    void updateImage(const CvArr* arr) CV_OVERRIDE;
```

Comment:
We have the same lines below.

The main difference is a base class: QOpenGLWidget vs QGLWidget
Use this approach:

```
#ifdef HAVE_QT6
typedef QOpenGLWidget OpenCVQtWidgetBase;
#else
typedef QGLWidget OpenCVQtWidgetBase;
#endif

class OpenGlViewPort : public OpenCVQtWidgetBase, public OCVViewPort
{
...
```

Reply:
This is better, reduces a lot of duplicate code.